src/vehicle.cpp

```diff
@@ -2660,6 +2683,28 @@ void Vehicle::ShowVisualEffect() const
 	if (sound) PlayVehicleSound(this, VSE_VISUAL_EFFECT);
 }
 
+/**
+ * Add penalty cargo-distance-traveled values for an idling vehicle.
+ * @param ticks How many ticks of idling to add penalty for
+ */
+void Vehicle::AddCargoPotentialPenalty(int ticks)
+{
+	if (_settings_game.economy.idle_loading_ct_penalty > 0) {
+		int assumed_speed;
+		if (this->type == VEH_ROAD) {
```

Comment:
definitely needs explanation about why different methods of speeds are used here. Or a common function

Reply:
I *think* just using Vehicle::GetDisplayMaxSpeed() is actually correct, except maybe aircraft would need to be scaled by the airplanes speed factor setting. Also noticed a bunch of missing override keywords for most vehicle types there, those should probably go in a separate PR.